packages/mui-base/src/TabsListUnstyled/useTabsList.ts

```diff
@@ -0,0 +1,204 @@
+import * as React from 'react';
+import { unstable_ownerDocument as ownerDocument } from '@mui/utils';
+import { isFragment } from 'react-is';
+import { useTabContext } from '../TabsUnstyled';
+import extractEventHandlers from '../utils/extractEventHandlers';
+
+const nextItem = (list: Element | null, item: Element | null): Element | null => {
+  if (!list) {
+    return null;
+  }
+
+  if (list === item) {
+    return list.firstChild as Element | null;
+  }
+  if (item && item.nextElementSibling) {
+    return item.nextElementSibling;
+  }
+  return list.firstChild as Element | null;
+};
+
+const previousItem = (list: Element | null, item: Element | null): Element | null => {
+  if (!list) {
+    return null;
+  }
+
+  if (list === item) {
+    return list.lastChild as Element | null;
+  }
+  if (item && item.previousElementSibling) {
+    return item.previousElementSibling;
+  }
+  return list.lastChild as Element | null;
+};
+
+const moveFocus = (
+  list: Element | null,
+  currentFocus: Element | null,
+  traversalFunction: (list: Element | null, currentFocus: Element | null) => Element | null,
+) => {
+  let wrappedOnce = false;
+  let nextFocus = traversalFunction(list, currentFocus);
+
+  while (list && nextFocus) {
+    // Prevent infinite loop.
+    if (nextFocus === list.firstChild) {
+      if (wrappedOnce) {
+        return;
+      }
+      wrappedOnce = true;
+    }
+
+    // Same logic as useAutocomplete.js
```

Comment:
This makes me wonder if we could reuse the useListbox hook in the future to control the behavior of selection.

Reply:
Let me check if everything necessary is already in the `useListbox` hook.

Reply:
Ah I didn't read the "future" part :) I think it's makes sense 👍